gradle-baseline-java/src/test/groovy/com/palantir/baseline/plugins/BaselineEnablePreviewFlagTest.groovy

```diff
@@ -156,19 +156,38 @@ class BaselineEnablePreviewFlagTest extends IntegrationSpec {
 
     def 'does not always apply'() {
         when:
-        File java8Dir = addSubproject('my-java-8-api', "sourceCompatibility = 8")
+        File java8Dir = addSubproject('my-java-8-api', """
+        sourceCompatibility = 8
+
+        tasks.withType(JavaCompile) {
+           doFirst {
+              println "Args for for \$name are \$options.allCompilerArgs"
+           }
+        } 
+        """.stripIndent())
+
         buildFile << '''
         subprojects {
             apply plugin: 'java-library'
             apply plugin: 'com.palantir.baseline-enable-preview-flag'
         }
         '''
-        writeSourceFileContainingRecord(java8Dir)
```

Comment:
`:compileJava` is failing here with:
```
error: cannot find symbol
|            public record Coordinate(int x, int y) {}
|                   ^
|        symbol:   class record
```

which makes sense to me as we are running with `sourceCompatibility = 8`. I think the output has changed from java 14 to java 15 and no longer contains the `use --enable-preview to enable records` line.

Changed the test to explicitly print the compiler args and check that `--enable-preview` is not set when running with `sourceCompatibility=8`.